File: app/src/main/java/com/tamil/smshelper/data/SmsTaskDao.java
package com.tamil.smshelper.data;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface SmsTaskDao {

    @Query("SELECT * FROM sms_task WHERE status = 0 ORDER BY id ASC")
    List<SmsTask> getPendingTasks();

    @Insert
    void insertTasks(SmsTask... tasks);

    @Update
    void updateTask(SmsTask task);

    @Query("UPDATE sms_task SET status = :status WHERE id = :taskId")
    void updateStatus(long taskId, int status);

    @Query("SELECT * FROM sms_task WHERE phoneNumber LIKE '%' || :keyword || '%'")
    List<SmsTask> searchByPhone(String keyword);

    @Delete
    void deleteTask(SmsTask task);

    @Query("DELETE FROM sms_task")
    void clearAll();

    @Insert
    void insertRecord(SendRecord record);

    @Update
    void updateRecord(SendRecord record);

    @Query("SELECT * FROM send_record WHERE phoneNumber = :phone ORDER BY timestamp DESC")
    List<SendRecord> getRecordsByPhone(String phone);

    @Query("SELECT * FROM send_record WHERE phoneNumber LIKE '%' || :keyword || '%' ORDER BY timestamp DESC")
    List<SendRecord> searchPhones(String keyword);

    @Query("SELECT phoneNumber, COUNT(*) as totalCount, " +
            "SUM(CASE WHEN success = 1 THEN 1 ELSE 0 END) as successCount, " +
            "SUM(CASE WHEN success = 0 THEN 1 ELSE 0 END) as failCount, " +
            "SUM(CASE WHEN replied = 1 THEN 1 ELSE 0 END) as replyCount " +
            "FROM send_record GROUP BY phoneNumber")
    List<PhoneStats> getPhoneStatistics();

    class PhoneStats {
        public String phoneNumber;
        public int totalCount;
        public int successCount;
        public int failCount;
        public int replyCount;
    }
}

File: app/src/main/java/com/tamil/smshelper/data/SmsTaskDao.kt
package com.tamil.smshelper.data

import androidx.room.*

@Dao
interface SmsTaskDao {

    @Query("SELECT * FROM sms_tasks WHERE status = 0 ORDER BY id ASC")
    fun getPendingTasks(): List<SmsTask>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertTasks(tasks: List<SmsTask>)

    @Update
    fun updateTask(task: SmsTask)

    @Query("UPDATE sms_tasks SET status = :status WHERE id = :taskId")
    fun updateStatus(taskId: Long, status: Int)

    @Query("DELETE FROM sms_tasks")
    fun clearAll()

    @Insert
    fun insertRecord(record: SendRecord)

    @Update
    fun updateRecord(record: SendRecord)

    @Query("SELECT * FROM sms_tasks WHERE phoneNumber LIKE '%' || :keyword || '%' ORDER BY id ASC")
    fun searchByPhone(keyword: String): List<SmsTask>

    @Delete
    fun deleteTask(task: SmsTask)

    @Query("SELECT * FROM send_records WHERE phoneNumber = :phone ORDER BY timestamp DESC")
    fun getRecordsByPhone(phone: String): List<SendRecord>

    @Query("SELECT phoneNumber, COUNT(*) AS totalCount, " +
            "SUM(CASE WHEN success THEN 1 ELSE 0 END) AS successCount, " +
            "SUM(CASE WHEN success THEN 0 ELSE 1 END) AS failCount, " +
            "SUM(CASE WHEN replied THEN 1 ELSE 0 END) AS replyCount " +
            "FROM send_records GROUP BY phoneNumber ORDER BY phoneNumber")
    fun getPhoneStatistics(): List<PhoneStats>

    @Query("SELECT phoneNumber, COUNT(*) AS totalCount, 0 AS successCount, 0 AS failCount, 0 AS replyCount FROM send_records WHERE phoneNumber LIKE '%' || :keyword || '%' GROUP BY phoneNumber")
    fun searchPhones(keyword: String): List<PhoneStats>

    data class PhoneStats(
        val phoneNumber: String,
        val totalCount: Int,
        val successCount: Int,
        val failCount: Int,
        val replyCount: Int
    )
}
